pramen/core/src/main/scala/za/co/absa/pramen/core/sql/SqlGeneratorMicrosoft.scala

```diff
@@ -97,7 +100,119 @@ class SqlGeneratorMicrosoft(sqlConfig: SqlConfig) extends SqlGeneratorBase(sqlCo
     }
   }
 
+  override def getAliasExpression(expression: String, alias: String): String = {
+    s"$expression AS ${escape(alias)}"
+  }
+
+  override def quote(identifier: String): String = {
+    validateIdentifier(identifier)
+    splitComplexIdentifier(identifier).map(quoteSingleIdentifier).mkString(".")
+  }
+
+  override def escape(identifier: String): String = {
+    if (needsEscaping(sqlConfig.identifierQuotingPolicy, identifier)) {
+      quote(identifier)
+    } else {
+      identifier
+    }
+  }
+
   private def getLimit(limit: Option[Int]): String = {
     limit.map(n => s"TOP $n ").getOrElse("")
   }
+
+  private def columnExpr(columns: Seq[String]): String = {
+    if (columns.isEmpty) {
+      "*"
+    } else {
+      columns.map(col => escape(col)).mkString(", ")
+    }
+  }
+
+  private def infoDateColumn: String = {
+    escape(sqlConfig.infoDateColumn)
+  }
+
+  private def quoteSingleIdentifier(identifier: String): String = {
+    val (escapeBegin, escapeEnd) = beginEndEscapeChars
+
+    if (
+      (identifier.startsWith(s"$escapeBegin") && identifier.endsWith(s"$escapeEnd")) ||
+        (identifier.startsWith(s"$escapeChar2") && identifier.endsWith(s"$escapeChar2"))
+    ) {
+      identifier
+    } else {
+      s"$escapeBegin$identifier$escapeEnd"
+    }
+  }
+
+  private[core] def splitComplexIdentifier(identifier: String): Seq[String] = {
+    val trimmedIdentifier = identifier.trim
+
+    if (trimmedIdentifier.isEmpty) {
+      throw new IllegalArgumentException(f"Found an empty table name or column name ('$identifier').")
+    }
+
+    val (escapeBegin1, escapeEnd1) = beginEndEscapeChars
+
+    val output = new ListBuffer[String]
+    val curColumn = new StringBuffer()
+    val len = trimmedIdentifier.length
+    var nestingChar = new ListBuffer[Char]
```

Comment:
Yes, I thought about it as well. But it seems `Stack` in Scala is slow, and deprecated.

https://www.scala-lang.org/api/2.12.19/scala/collection/mutable/Stack.html

Reply:
I was thinking implementing my own stack based on `ListBuffer`. I discarded the idea since it is only for one small use case. But maybe it is worth it, for cleanness. What do you think?